### PR TITLE
ZCS-4846: Upgrade to Apache HttpClient to 4.x

### PR DESCRIPTION
**Pull request by jdavid-synacor**
Fixed code to use new httpclient lib
Fixed compilation errors

test:
     [echo] /home/zimbra/repos/zm-certificate-manager-store/src/java-test not found. Will not run unit tests

### CODE REVIEW

**Comment by jdavid-synacor** [comment]
I got some warnings.


compile:
    [javac] Compiling 15 source files to /home/zimbra/repos/zm-certificate-manager-store/build/classes
    [javac] /home/zimbra/repos/zm-certificate-manager-store/src/java/com/zimbra/cert/MyExportPrivKey.java:34: warning: BASE64Encoder is internal proprietary API and may be removed in a future release
    [javac] import sun.misc.BASE64Encoder;
    [javac]                ^
    [javac] /home/zimbra/repos/zm-certificate-manager-store/src/java/com/zimbra/cert/MyExportPrivKey.java:60: warning: BASE64Encoder is internal proprietary API and may be removed in a future release
    [javac]             BASE64Encoder encoder=new BASE64Encoder();
    [javac]             ^
    [javac] /home/zimbra/repos/zm-certificate-manager-store/src/java/com/zimbra/cert/MyExportPrivKey.java:60: warning: BASE64Encoder is internal proprietary API and may be removed in a future release
    [javac]             BASE64Encoder encoder=new BASE64Encoder();
    [javac]                                       ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: /home/zimbra/repos/zm-certificate-manager-store/src/java/com/zimbra/cert/OutputParser.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 3 warnings

BUILD SUCCESSFUL
Total time: 4 seconds

**Comment by rupalid** [comment]
You can ignore the warnings

File: ivy.xml [hunk 1]
```diff
@@ -8,7 +8,11 @@
   <dependency org="javax.servlet" name="javax.servlet-api" rev="3.1.0" />
   <dependency org="dom4j" name="dom4j" rev="1.5.2" />
   <dependency org="com.google.guava" name="guava" rev="23.0" />
-  <dependency org="commons-httpclient" name="commons-httpclient" rev="3.1" />
+  <dependency org="org.apache.httpcomponents" name="httpclient" rev="4.5.5"/>
```
Comment (rupalid) [inline ivy.xml:11]: Again import the jars that are needed.

**Review by rupalid — APPROVED** [review]

**Review by chinmay15 — APPROVED** [review]
LGTM

**Pull request merged**